### DISCUSSION TITLE
Doubt about cc2652R coordinator firmware upgrade v20230507

### DISCUSSION THREAD

**Discussion by kitopopo** [Koenkk/zigbee2mqtt#18774] [Devices]
Good morning dear friend @Koenkk , 

I have a question about the upgrade of the coordinator cc2652R. I am currently using home assistant and I wanted to upgrade the coordinator since I haven't updated it for 2 years. I would like to know if I need to make a copy of the nvram coordinator since I found this text in the instructions and I have doubts:

**Please note that if you are a zigpy/ZHA user, you might want to take a backup of your NVRAM data before updating your stick. Information on this can be found on the [zigpy-znp documentation](https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md#backup-and-restore).**

I know that all the information about devices added to my zigbee2mqtt network is stored on my ssd inside my docker container and I think that making a backup of my ssd would be enough to avoid having to repair all the devices but i am not sure. 
After the firmware upgrade, will I lose all the devices in h.a. and will I have to add them to the network again? Will I also lose all my home assistant entities? 

I have zigbee2mqtt and home assistant installed in containers normally but I don't know if I use zigpy. I thank you in advance for your answer since I am afraid of making an upgrade. I think that when I used domoticz I updated the coordinator without problems. thank you. Kind regards.

**Comment by francisp2** [comment]
Nothing should change after the firmware upgrade. All your devices will still be paired.

**Reply by kitopopo** [reply]
Thanks for your answer, I think I remember that a few years ago when I was using domoticz nothing changed after upgrade my cc2652r and everything remained the same after the update, but the comment in the documentation caught my attention that I didn't quite understand then why these 2 copies are used: network backup and Nvram Backup:

https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md#linux

Thanks in advanced

**Comment by kitopopo** [comment]
Hi dear @Koenkk , can you confirm? Thanks in advanced

**Reply by Koenkk** [reply]
You can just update, all entities will persist

**Comment by kitopopo** [comment]
Thanks @francisp2 @Koenkk . I will try. Best regards

**Comment by kitopopo** [comment]
Dear friend @Koenkk , it was impossible burn the coordinator firmware from 2023 year with the software TI flash programmer 2 but is possible burn the firmware from 2022 year with TI flash programmer 2.
I could only record the 2023 coordinator firmware in my cc2652R with the zigstar multitool software. Is this normal? Now we need use the zigstar instead ti flash programmer? I also tried with uniflash 8.0.0 but my adapter is not recognized. 
With zigstar the stick cc2652R works correctly with firmware 05/2023. Thanks in advanced. Best regards

**Reply by Koenkk** [reply]
see https://github.com/Koenkk/Z-Stack-firmware/issues/468#issuecomment-1673523085
